Documentation/faq.md

```diff
@@ -118,6 +118,12 @@ Every new etcd cluster generates a new cluster ID based on the initial cluster c
 
 Usually this warning happens after tearing down an old cluster, then reusing some of the peer addresses for the new cluster. If any etcd process from the old cluster is still running it will try to contact the new cluster. The new cluster will recognize a cluster ID mismatch, then ignore the request and emit this warning. This warning is often cleared by ensuring peer addresses among distinct clusters are disjoint.
 
+#### What does the etcd warning "snapshotting is taking more than x seconds to finish ..." mean?
+
+etcd can transfer its key value store as a snapshot for [backup][backup] or to a slow follower for bring it up to speed. The expected transfer time is based on transferring snapshot over a 1 gigabit/s connection assuming a minimum tcp throughput of 100 MB/s. When the actual transfer time is greater than min(30 seconds, expected transfer time), the warning appears. The warning allows one to reason about the etcd snapshotting performance. If one expects a slow snapshotting performance due to a slow underlying infrastructure, then ignore this warning. If the performance is slower than expected, use this warning for debugging/investigating. 
```

Comment:
etcd sends a snapshot of its complete key-value store to refresh slow followers and for [backups][backup]. Slow snapshot transfer times increase MTTR; if the cluster is ingesting data with high throughput, slow followers may livelock by needing a new snapshot before finishing receiving a snapshot. To catch slow snapshot performance, etcd warns when sending a snapshot takes more than thirty seconds and exceeds the expected transfer time for a 1Gbps connection.